File: in_Java/0821 Shortest Distance to a Character.java
class Solution {
    public int[] shortestToChar(String S, char C) {
        int[] res = new int[S.length()];
        List<Integer> L = new ArrayList<>();
        for(int i = 0; i < S.length(); i++){
            if(S.charAt(i) == C) L.add(i);
        }
        for(int i = 0; i < S.length();i++){
            int cur = Integer.MAX_VALUE;
            for(int x : L){
                if(Math.abs(x-i) < cur) cur = Math.abs(x-i);
            }
            res[i] = cur;
        }
        return res;
    }
}

File: in_Python/0821 Shortest Distance to a Character.py
class Solution(object):
    def shortestToChar(self, S, C):
        """
        :type S: str
        :type C: str
        :rtype: List[int]
        """
        c_id = []
        for i in range(len(S)):
            if S[i] == C:
                c_id.append(i)
        res = []
        for i in range(len(S)):
            res.append(len(S))
        for cid in c_id:
            for i in range(len(res)):
                res[i] = min(res[i], abs(cid-i))
        return res
        